File: C#/第一部分 C#语言/第二章 顺序结构程序设计/第五节 数据输入输出/io.cs
// 由bhw设计
/*
第2章 第5节 数据输入输出

一、输入输出基础
1) 输出：Console.Write / Console.WriteLine
2) 输入：Console.ReadLine() 读取一整行字符串
3) 类型转换：int.Parse / double.Parse / TryParse

二、原理
1) 控制台输入本质是字符串。
2) 你要根据目标类型做解析（例如转 int）。
3) 初学者优先使用 TryParse 处理错误输入。

三、习题
1) 输入姓名和年龄并输出。
2) 输入两个整数并输出和。

四、答案与讲解
1) name = Console.ReadLine();
2) int.TryParse 可避免异常中断。
*/
using System; // 控制台输入输出

class Program // 程序类
{
    static void Main() // 主函数
    {
        Console.Write("请输入姓名: "); // 输出输入提示
        string name = Console.ReadLine() ?? ""; // 读取姓名，防空值
        Console.Write("请输入年龄: "); // 输出年龄提示
        string ageText = Console.ReadLine() ?? "0"; // 读取年龄文本

        if (!int.TryParse(ageText, out int age)) // 尝试把文本转 int
        {
            Console.WriteLine("年龄输入格式错误"); // 解析失败提示
            return; // 结束程序
        }

        Console.WriteLine($"name={name}, age={age}"); // 输出解析结果
    } // Main 结束
} // 类结束
// 由bhw设计

File: C++/第一部分 C++语法/第二章 顺序结构程序设计/第三节 标准输入输出/io.cpp
// 由bhw设计
/*
第二章 第三节 标准输入输出

一、输入输出对象
1) cout：标准输出，向屏幕打印信息。
2) cin：标准输入，从键盘读取数据。
3) endl：换行并刷新缓冲区。

二、输入注意事项
1) 按空格分隔时，cin会按变量类型依次读取。
2) 类型不匹配会导致读取失败。
3) 对字符串整行输入可用getline（后续章节展开）。

三、练习题
1) 输入姓名和年龄并输出自我介绍。
2) 输入两个整数，输出它们的和与差。
*/
#include <iostream> // 输入输出头文件
#include <string> // 字符串头文件
using namespace std; // 标准命名空间

int main() // 主函数
{
    string name; // 定义字符串变量存姓名
    int age = 0; // 定义整型变量存年龄
    int x = 0; // 定义第一个整数
    int y = 0; // 定义第二个整数

    cout << "请输入姓名和年龄（如 bhw 18）: "; // 提示用户输入
    cin >> name >> age; // 读取姓名和年龄
    cout << "你好，" << name << "，你今年" << age << "岁。" << endl; // 输出问候

    cout << "请输入两个整数: "; // 提示输入两个整数
    cin >> x >> y; // 读取x和y
    cout << "和 = " << x + y << endl; // 输出和
    cout << "差 = " << x - y << endl; // 输出差

    return 0; // 程序结束
} // main结束

/*
练习题答案（注释版）
1) 自我介绍：
// string name; // 姓名
// int age; // 年龄
// cin >> name >> age; // 读入
// cout << "我叫" << name << "，今年" << age << "岁。" << endl; // 输出

2) 和与差：
// int a, b; // 定义两个变量
// cin >> a >> b; // 输入
// cout << (a + b) << endl; // 输出和
// cout << (a - b) << endl; // 输出差
*/
// 由bhw设计
